### ISSUE TITLE
Code Issue?

### ISSUE THREAD

**Issue by osivaz61** [XiaofeiWang2018/DR#3]
extractor = ModelOutputs(cls_net)
                features, fc_output = extractor(cls_input)
                cam_all_sr = {'size256': [], 'size128': [], 'size64': [], 'size32': []}
                for batch in range(fc_output.shape[0]):
                    device = fc_output[batch].device.index
                    index = np.argmax(fc_output[batch].cpu().data.numpy())
                    if index == 0:
                        one_hot = np.zeros((1, fc_output[batch].size()[-1]), dtype=np.float32)
                        one_hot[0][0] = 1
                    if index > 0:
                        one_hot = np.ones((1, fc_output[batch].size()[-1]), dtype=np.float32)
                        one_hot[0][0] = 0
                    if batch == 0:
                        one_hot_all = Variable(torch.from_numpy(one_hot).cuda(device), requires_grad=True)
                    else:
                        one_hot_all = torch.cat((Variable(torch.from_numpy(one_hot).cuda(device), requires_grad=True), one_hot_all), 0)

if the index of batches is the same, there is no problem. But if they are different, it calculates the opposite loss between one_hot_all and fc_output.

Example:

fc_output                          one_hot_all
[[0.9, 0.1, 0.1, 0.1, 0.1]      [[1, 0, 0, 0, 0]
 [0.8, 0.2, 0.2, 0.3, 0.4]]      [1, 0, 0, 0, 0]]            There is no problem.

fc_output                          one_hot_all
[[0.2, 0.1, 0.1, 0.1, 0.8]       [[0, 1, 1, 1, 1]
 [0.3, 0.2, 0.7, 0.3, 0.4]]       [0, 1, 1, 1, 1]]           There is no problem.

fc_output                          one_hot_all
[[0.9, 0.1, 0.1, 0.1, 0.7]       [[0, 1, 1, 1, 1]
 [0.3, 0.2, 0.4, 0.9, 0.4]]       [1, 0, 0, 0, 0]]           There is problem here because of this line torch.cat((Variable(torch.from_numpy(one_hot).cuda(device), requires_grad=True), one_hot_all), 0).

should be one_hot_all = torch.cat((one_hot_all, Variable(torch.from_numpy(one_hot).cuda(device), requires_grad=True)), 0) ???

Thanks
